Mesh_3/include/CGAL/IO/facets_in_complex_3_to_triangle_mesh.h

```diff
@@ -71,59 +67,120 @@ void facets_in_complex_3_to_triangle_mesh(const C3T3& c3t3, TriangleMesh& graph)
   typedef CGAL::Hash_handles_with_or_without_timestamps                  Hash_fct;
   typedef boost::unordered_map<Vertex_handle, std::size_t, Hash_fct>     VHmap;
 
-  typedef CGAL::cpp11::array<std::size_t, 3>                             Face;
+  typedef typename C3T3::size_type                                       size_type;
 
-  typename std::iterator_traits<Ficit>::difference_type nf =
-    std::distance(c3t3.facets_in_complex_begin(), c3t3.facets_in_complex_end());
-
-  std::vector<Face> faces;
-  faces.reserve(nf);
-  std::vector<Point_3> points;
-  points.reserve(nf / 2); // approximating Euler
+  size_type nf = c3t3.number_of_facets_in_complex();
+  faces.reserve(faces.size() + nf);
+  points.reserve(points.size() + nf/2); // approximating Euler
 
   VHmap vh_to_ids;
   std::size_t inum = 0;
 
+  // Whether all the facets of the c3t3 are being exported, or only those of a particular subdomain
+  const bool export_all_facets = (sd_index == -1);
+
   for(Ficit fit = c3t3.facets_in_complex_begin(),
             end = c3t3.facets_in_complex_end(); fit != end; ++fit)
   {
     Cell_handle c = fit->first;
     int s = fit->second;
-    Face f;
+    Face f(3);
```

Comment:
see for example `read_OFF_internal::resize()`. Maybe a global more standard solution should be found.